packages/node/src/utils/project.ts

```diff
@@ -62,3 +85,159 @@ export function isCustomHandler<K extends string, F>(
 ): handler is SubqlCustomHandler<K, F> {
   return !isBaseHandler(handler);
 }
+
+async function syncRemoteFileToLocal(
+  reader: Reader,
+  root: string,
+  inputFile: string,
+  ext?: string,
+  content?: string,
+): Promise<string> {
+  const res = content ?? (await reader.getFile(inputFile));
+  const name = blake2AsHex(inputFile).substr(2, 10);
+  const fileName = ext ? `${name}${ext}` : name;
+  const outputPath = path.resolve(root, fileName);
+  await fs.promises.writeFile(outputPath, res as string);
+  return outputPath;
+}
+
+export async function updateDataSources(
+  _dataSources:
```

Comment:
I suggest you make separate `updateDataSources` for each manifest versions, and put them together so next time we add a new version we know what need to implement.